compiler/lib/src/main/scala/analysis/CheckSemantics/CheckUses.scala

```diff
@@ -121,24 +121,63 @@ object CheckUses extends UseAnalyzer {
     a.dictionaryNeeded match {
       case true => {
         val impliedTypeUses = List(
-          "FwChanIdType", 
-          "FwEventIdType", 
-          "FwOpcodeType", 
-          "FwPacketDescriptorType", 
-          "FwTlmPacketizeIdType"
+          List("FwChanIdType"),
+          List("FwDpIdType"),
+          List("FwDpPriorityType"),
+          List("FwEventIdType"),
+          List("FwOpcodeType"),
+          List("FwPacketDescriptorType"),
+          List("FwSizeType"),
+          List("FwSizeStoreType"),
+          List("FwTimeBaseStoreType"),
+          List("FwTimeContextStoreType"),
+          List("FwTlmPacketizeIdType")
+        )
+        val impliedEnumUses = List(
+          List("Fw", "DpState"),
+          List("Fw", "DpCfg", "ProcType")
+        )
+        val impliedConstantUses = List(
+          List("Fw", "DpCfg", "CONTAINER_USER_DATA_SIZE")
         )
         val (_, node1, _) = node
-        val mapping = a.nestedScope.get (NameGroup.Type) _
-        for {
-          a <- Result.foldLeft (impliedTypeUses) (a) ((a, t) => {
+        def checkImpliedUses(
+          impliedUses: List[List[String]],
+          id: AstNode.Id,
+          typeName: String,
+          ng: NameGroup
+        ): Result.Result[Set[Symbol]] = {
+          Result.foldLeft (impliedUses) (Set[Symbol]()) ((s, identList) => {
```

Comment:
I have one more change to suggest here. Instead of using Set[Symbol] to represent these concepts, can we use Set[Symbol.Type], Set[EnumSymbol], and Set[Constant]? That would be more precise. And can we could throw an error if, for example, the symbol `Fw.DpState` is defined, but it's not an enum symbol? If we don't do this, I'm concerned we won't have enough error checking. For example, if `Fw.DpState` were be a struct type definition, this check would pass.